File: Backend/src/Sienna.Data/Models/EspressoShot.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sienna.Data.Models
{
    public class EspressoShot
    {
        [Key]
        public Guid Id { get; set; }

        [StringLength(128)]
        public string? BeanType { get; set; }

        public double Grind { get; set; }

        public int Beans { get; set; }

        public int Pressure { get; set; }

        public int Water { get; set; }

        public int BrewTime { get; set; }

        public int Flavour { get; set; }

        public int Rating { get; set; }

        [StringLength(256)]
        public string? Comments { get; set; }
    }
}

File: Frontend/src/app/models/espresso-shot.ts
export enum Flavour {
    Acidic = 1,
    SlightlyAcidic = 2,
    BalancedFlavour = 3,
    SlightlyBitter = 4,
    Bitter = 5
}

export interface EspressoShot {
    id: string;
    beanType?: string;
    grind: number;
    beans: number;
    pressure: number;
    water: number;
    brewTime: number;
    flavour: number;
    rating: number;
    comments?: string;
}

export interface EspressoShotDto {
    beanType?: string;
    grind: number;
    beans: number;
    pressure: number;
    water: number;
    brewTime: number;
    flavour: number;
    rating: number;
    comments?: string;
}